record/custom_records.go

```diff
@@ -22,3 +27,62 @@ func (c CustomSet) Validate() error {
 
 	return nil
 }
+
+// Encode writes the CustomSet to the provided io.Writer in big endian format.
+func (c *CustomSet) Encode(w io.Writer) error {
+	for key, value := range *c {
+		// Write the map key.
+		if err := binary.Write(w, binary.BigEndian, key); err != nil {
+			return err
+		}
+
+		// Write the length of the map value.
+		valueLength := uint16(len(value))
+		err := binary.Write(w, binary.BigEndian, valueLength)
+		if err != nil {
+			return err
+		}
+
+		// Write the map value.
+		if _, err := w.Write(value); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// Decode reads the CustomSet from the provided io.Reader in big endian format.
+func (c *CustomSet) Decode(r io.Reader) error {
+	if *c == nil {
+		*c = make(CustomSet)
+	}
+
+	for {
+		// Read the map key.
+		var key uint64
+		if err := binary.Read(r, binary.BigEndian, &key); err != nil {
+			if errors.Is(err, io.EOF) {
+				break
+			}
+
+			return err
+		}
+
+		// Read the length of the map value.
+		var valueLength uint16
+		err := binary.Read(r, binary.BigEndian, &valueLength)
```

Comment:
So in the actual TLV format, we don't use a fixed size integer here. Instead, we use a variable sized it, big endian, we call `BigSize`: https://github.com/lightningnetwork/lnd/blob/7fb233326eec1cd47fc7cfe263210e058c2a7171/tlv/stream.go#L92-L102

Reply:
Same goes for the above when we read.